.github/workflows/test-python-client.yml

```diff
@@ -0,0 +1,100 @@
+      matrix:
+        python-version: [ "3.8", "3.9", "3.10" ]
+        os: [ubuntu-latest]
+        experimental: [false]
+        include:
+          - python-version: 3.11
+            os: ubuntu-latest
+            experimental: true
+          - python-version: 3.10
+            os: macos-latest
+            experimental: false
+          - python-version: 3.10
+            os: windows-latest
+            experimental: false
+    continue-on-error: ${{ matrix.experimental }} # https://ncorti.com/blog/howto-github-actions-build-matrix
+    steps:
+
+      - name: Extract branch name
+        shell: bash
+        run: echo "branch=$(echo ${GITHUB_REF#refs/heads/})" >> $GITHUB_OUTPUT
+        id: extract_branch
+
+      - name: Checkout code
+        uses: actions/checkout@v3.3.0
+
+      # - name: Cache Python dependencies
+      #   uses: actions/cache@v3
+      #   with:
+      #     path: python-client/.venv
+      #     key: ${{ runner.os }}-${{ matrix.python-version }}-client-${{ hashFiles('python-client/pdm.lock')}}
+      #     restore-keys: ${{ runner.os }}-client
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run Python commands
+        run: |
+          python3 -m pip install virtualenv -U --upgrade pip
+          echo "Python ($(which python)) packages:\n $(python -m pip list)"
+
+      - name: Setup Gradle # To cache ~/.gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: false
+
+      - name: Set up JDK 17 # Needed because of Gradle anyway
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Set up Pandoc (needed for python-client doc)
+        uses: r-lib/actions/setup-pandoc@v2
+        with:
+          pandoc-version: '3.1.7' # https://github.com/jgm/pandoc/releases
+  
+      # - name: Cache Giskard test resources
+      #   uses: actions/cache@v3
+      #   with:
+      #     path: ~/.giskard
+      #     key: ${{ runner.os }}-giskard-test-resources-${{ hashFiles('python-client/tests/fixtures/**/*py')}}
+      #     restore-keys: ${{ runner.os }}-giskard-test-resources
+
+      # - name: Get number of available workers
+      #   id: nb_workers
+      #   # run: echo "nb_workers=1" >> $GITHUB_OUTPUT
+      #   run: python3 -c 'import os; print(f"nb_workers={max(1,len(os.sched_getaffinity(0))-1)}")' >> $GITHUB_OUTPUT
+
+      - name: Install deps
+        run: ./gradlew :python-client:install --info --parallel
+
+      - name: Build only
+        run: ./gradlew :python-client:build --info --parallel
+
+      - name: Test
+        env:
+          PYTEST_XDIST_AUTO_NUM_WORKERS: 1 #TODO(Bazire): put back dynamic number
+        run: ./gradlew :python-client:test-all --info --parallel
+
+      - name: Package
+        run: ./gradlew :python-client:package --info --parallel
+
+      - name: Lint
```

Comment:
linting should happen first because it's a lot faster compared to tests execution. In case there are linting issues it'll fail fast